### PR TITLE
[WIP] Development and deployment flow description

### PR DESCRIPTION
**Pull request by JirkaChadima**
I am going to describe how we should develop and deploy our repositories. Feel free to discuss, it's a heavy work in progress and we will need to solve a lot of problems before finalizing this.

This pull request will solve this card https://trello.com/c/DaZPaJU8/11-github-milestones-issues-describe-development-workflow (it's private forr now, we're experimenting with project management tools)

### CODE REVIEW

**Comment by JirkaChadima** [comment]
@kvbik @czervenka @AugustoL @adklempner Please, have a look at this. We should somehow formalize and describe our development/release process and then stick to it. This is an attempt to do so.

I'm of course open to discussion and changes, the process should be based on our mutual agreement. My goal is to minimize manual work, enable contributions from anyone, and automate all the things possible :-)

**Comment by JirkaChadima** [comment]
@czervenka @MatiasOS Do you guys have any comments on this?

**Comment by kvbik** [comment]
re `feature/155-add-something-feature` it is not always the case the PR refers to a specific issue, not a bad idea, but i wouldn't be that strict, i guess...

File: development.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+If you want to contribute, follow the contributing guidelines in each repository.
+In general, if you find a problem or have an idea, start a GitHub issue. If you
+have some code ready, feel free to create a Pull request, usually into a `develop`
+branch. Thank you!
+
+Every pull request has to be approved by one of our core members and only after
+an approval can be merged.
+
+For now we don't enforce any coding style, but that is probably going to change
+at some point. Just keep your formatting sane or at least similar to the
```
Comment (czervenka) [inline development.md]: ... formatting **same** or at least similar ...

**Comment by JirkaChadima** [comment]
I think marking branches with issue numbers is nice, but I'd rather keep it optional. You don't always have the issue...

File: development.md (hunk 1 above)
Comment (JirkaChadima) [reply development.md]: It's actually meant to say `sane` :)

**Pull request merged**